<launch>
    <arg name="map_name" default="$(env STUDENT_PRACTICE)/maps/ifl/ifl1.bt"/>

    <node pkg = "normal_extractor" type = "normal_extractor_node" name = "normal_extractor_node" output="screen">
        <!-- frame id -->
        <param name = "frame_id" type = "string" value = "map" />
        <!-- max z to extract the normal-->
        <param name = "filter_z" value = "3" />
        <!-- mode 1: pcl_estimator mode 2: octree_estimator -->
        <param name = "mode" value = "1" />
        
        <!-- pcl estimator -->
        <!-- search radius for pcl normal estimator -->
        <param name = "resolution_pcl" value = "0.03" />
        
        <!-- octree estimator -->
        <!-- leaf resolution -->
        <param name = "resolution_tree" value = "0.05" />
        <!-- use the normal of which tree depth -->
        <param name = "tree_depth" value = "14" />


        <!-- visualize -->
        <param name = "visualize" type = "bool" value = "true" />
        <!-- display density -->
        <param name = "mod" value = "20" />
        <!-- duration for visualize -->
        <param name = "duration" value = "20" />


        <remap from = "~/cloud_in" to = "/octomap_point_cloud_centers" />
        <remap from = "~/sensor_cloud_in" to = "/sensor_cloud_in" />
    </node>
    
    <node pkg="octomap_server" type="octomap_server_node" name="octomap" args="$(arg map_name)"/>

    <node pkg="rqt_reconfigure" type="rqt_reconfigure" name="reconfigure" />

    <node pkg="rviz" type="rviz" name="rviz_normal" args="-d $(find normal_extractor)/launch/normal.rviz" />
</launch>
